### DISCUSSION TITLE
what does this log in console means - Found pattern XYZ in script ?

### DISCUSSION THREAD

**Discussion by pierrelasse** [YimMenu/YimMenu#2154] [Q&A] [answered]
`[DEBUG/script_function.cpp:35] Found pattern SFSA in script freemode`

**Comment by lonelybud** [comment]
yes need to know - 

`[17:55:01.3448327][DEBUG/script_function.cpp:35] Found pattern RSD in script PauseMenu_Multiplayer`

**Comment by lonelybud** [comment]
@pierrelasse please give good title to this discussion

**Reply by pierrelasse** [reply]
give me a good title then

**Reply by lonelybud** [reply]
what does this log in console means - Found pattern XYZ in script ?

**Comment by lonelybud** [comment]
@gir489returns @Yimura this is not something to worry about right?

**Reply by Yimura** [reply]
You can ignore this, it's just the menu working as intended. If it was something problematic it would be a WARN or FATAL level log.
If you're really keen in understanding it's best to look at the code.

**Reply by lonelybud** [reply]
thanks. @pierrelasse you can close the discussion now.

**Reply by pierrelasse** [reply]
so the answer is: search for yourself in the code?

**Reply by Yimura** [reply]
Doubt about what? What is the actual question, what it means? What it's used for?

**Reply by pierrelasse** [reply]
Yes, what does it mean? What causes it?

**Reply by lonelybud** [reply]
there are some script functions defined in this file -
[`src\script_function.hpp`](https://github.com/YimMenu/YimMenu/blob/master/src/script_function.hpp)

whenever they need to be executed then they are called at that time may be like this
```
script_function::populate_ip(); // populate if not populated

script_function::call();
```

now my doubt is that `location.value()` should return `0` when `get_code_location_by_pattern` returns `std::nullopt` in order for code to work properly

https://github.com/YimMenu/YimMenu/blob/b28ce2188d4987a75e25fb067d1a40b257ab6717/src/script_function.cpp#L37C4-L37C9

**Reply by pierrelasse** [reply]
So the message is cause here but idfk what this does / is. Maybe it has to do with disabling the anticheat? Idk
![image](https://github.com/YimMenu/YimMenu/assets/76223390/d48eb098-85fc-49b6-9753-e1369cb83ab9)

**Reply by lonelybud** [reply, marked as answer]
no as i mentioned above it is not related to anticheat. I am pretty sure the code is used for calling scripts for example when `dance mod` is turned on or for example you join someones ceo using the `join ceo` button on player page. As a way when doing It logs also logs if the pattern is found in memory.

**Reply by lonelybud** [reply]
i wish that it should work properly if pattern is not found.
